rust/index/src/fulltext/types.rs

```diff
@@ -92,60 +93,71 @@ impl<'me> FullTextIndexWriter<'me> {
 
     async fn populate_frequencies_and_posting_lists_from_previous_version(
         &self,
-        token: &str,
+        tokens: &Vec<Token>,
```

Comment:
this is slightly weird, but the tantivy token type is already leaking out of our abstraction and this avoids having to remap the tokens to strs
not strongly opinionated here, ok going the remapping route